pkg/scheduler/objects/queue_test.go

```diff
@@ -2594,3 +2594,19 @@ func TestQueue_allocatedResFits(t *testing.T) {
 		})
 	}
 }
+
+func TestQueueSetMaxRunningApps(t *testing.T) {
+	queue := &Queue{}
+	maxApps := uint64(10)
+
+	queue.SetMaxRunningApps(maxApps)
+	assert.Equal(t, maxApps, queue.maxRunningApps)
+
+	queue = nil
+	queue.SetMaxRunningApps(maxApps)
+	defer func() {
```

Comment:
The defer section should come earlier, otherwise it will not catch the panic

Reply:
Addressed in latest PR, thanks!